02_classify_mutations/nbconverted/explore_shuffled_cancer_type.py

```diff
@@ -0,0 +1,219 @@
+].copy()
+
+new_compressed_results_df = au.load_compressed_prediction_results(new_results_dir, 'gene')
+new_compressed_results_df = new_compressed_results_df[
+    (new_compressed_results_df.training_data.isin(['me_27k', 'me_450k'])) &
+    (new_compressed_results_df.n_dims == 5000)
+].copy()
+
+old_results_df['n_dims'] = 'raw'
+old_results_df = pd.concat((old_results_df, old_compressed_results_df))
+new_results_df['n_dims'] = 'raw'
+new_results_df = pd.concat((new_results_df, new_compressed_results_df))
+
+
+# In[4]:
+
+
+print(old_results_df.seed.unique())
+print(old_results_df.training_data.unique())
+print(old_results_df.n_dims.unique())
+print(old_results_df.shape)
+old_results_df.head()
+
+
+# In[5]:
+
+
+print(new_results_df.seed.unique())
+print(new_results_df.training_data.unique())
+print(new_results_df.n_dims.unique())
+print(new_results_df.shape)
+new_results_df.head()
+
+
+# In[6]:
+
+
+old_shuffled_df = (
+    old_results_df[(old_results_df.signal == 'shuffled') &
+                   (old_results_df.data_type == 'test')]
+).sort_values(by=['identifier', 'training_data', 'seed', 'fold'])
+new_shuffled_df = (
+    new_results_df[(new_results_df.signal == 'shuffled') &
+                   (new_results_df.data_type == 'test')]
+).sort_values(by=['identifier', 'training_data', 'seed', 'fold'])
+
+shuffled_diff_df = old_shuffled_df.copy()
+shuffled_diff_df['auroc_diff'] = old_shuffled_df['auroc'] - new_shuffled_df['auroc']
+shuffled_diff_df['aupr_diff'] = old_shuffled_df['aupr'] - new_shuffled_df['aupr']
+shuffled_diff_df = (shuffled_diff_df
+    .drop(columns=['auroc', 'aupr', 'signal', 'experiment', 'n_dims', 'data_type'])
+    .groupby(['identifier', 'training_data'])
+    .mean()
+    .reset_index()
+    .drop(columns=['seed', 'fold'])
+)
+
+print(shuffled_diff_df.shape)
+shuffled_diff_df.head()
+
+
+# In[7]:
+
+
+metric = 'aupr'
+num_examples = 20
+
+top_df = (shuffled_diff_df
+    .sort_values(by='{}_diff'.format(metric), ascending=False)
+    .head(num_examples)
+)
+bottom_df = (shuffled_diff_df
+    .sort_values(by='{}_diff'.format(metric), ascending=False)
+    .tail(num_examples)
+)
+
+plot_df = pd.concat((top_df, bottom_df)).reset_index()
+
+sns.set({'figure.figsize': (24, 5)})
+sns.barplot(data=plot_df, x=plot_df.index, y='{}_diff'.format(metric),
+            hue='training_data',  dodge=False,
+            hue_order=sorted(plot_df.training_data.unique()))
+
+def show_values_on_bars(ax):
+    for i in range(plot_df.shape[0]):
+        _x = i
+        _y = plot_df.loc[i, '{}_diff'.format(metric)]
+        val = plot_df.loc[i, 'identifier']
+        if _y > 0:
+            ax.text(_x, _y + 0.01, val, ha="center") 
+        else:
+            ax.text(_x, _y - 0.02, val, ha="center") 
+        
+show_values_on_bars(plt.gca())
+plt.gca().get_xaxis().set_visible(False)
+plt.ylabel('{} difference'.format(metric.upper()))
+plt.title('Performance difference between new and old shuffling scheme', size=14)
```

Comment:
The results sort-of make sense to me - in addition to changing the shuffling scheme, I also slightly changed the way that the PCA decompositions are calculated for the methylation datasets. Because of this I expected to see more of the positive changes in the methylation-based classifiers (which is the case), but I expected the magnitude of the changes to be much less than the negative changes induced by the change to the shuffling (which are roughly evenly spread across data types).

I haven't looked at a baseline where we just vary the seed, but that's a good idea. This notebook was really just intended as a sanity check that we weren't seeing anything crazy when we changed the shuffling scheme, so I think in that respect I'm okay with these results.